jcl-over-slf4j/src/main/java/org/apache/commons/logging/impl/SLF4JLogFactory.java

```diff
@@ -56,15 +56,15 @@
 
   /**
    * The {@link org.apache.commons.logging.Log}instances that have already been
-   * created, keyed by logger name.
+   * created, keyed by Logger instance.
    */
-  ConcurrentMap<String, Log> loggerMap;
+  ConcurrentMap<Logger, Log> loggerMap;
```

Comment:
What happens when the webapp the loaded Logger is unloaded? loggerMap will still retain a reference, no?

Maybe use WeakHashMap with a ReadWriteLock to control access.